constraints/miden-vm/decoder.air

```diff
@@ -0,0 +1,594 @@
+    # Get value of the f_push flag
+    let f_push = get_f_push(op_bits, extra)
+
+    # Values of the rows to be added to the op group table when a SPAN or a RESPAN operation is 
+    # executed.
+    let v_i = $alpha[0] + $alpha[1] * addr' + $alpha[2] * (group_count - i) + $alpha[3] * hasher[i]
+
+    # Row value for group in hasher[1] to be added to the op group table when a SPAN or a RESPAN 
+    # operation is executed.
+    let v_1 = $alpha[0] + $alpha[1] * addr' + $alpha[2] * (group_count - 1) + $alpha[3] * hasher[1]
+
+    let prod_v_3 = prod([$alpha[0] + 
+                         $alpha[1] * addr' + 
+                         $alpha[2] * (group_count - i) + 
+                         $alpha[3] * hasher[i] for i in 1..4])
+
+    let prod_v_7 = prod([$alpha[0] + 
+                         $alpha[1] * addr' + 
+                         $alpha[2] * (group_count - i) + 
+                         $alpha[3] * hasher[i] for i in 1..8])
+
+    # The value of the row to be removed from the op group table.
+    let u = $alpha[0] + $alpha[1] * addr + $alpha[2] * group_count + $alpha[3] * 
+        ((hasher[0]' * 2^7 + op_index') * (1 - f_push + s0' * f_push)) = 0
+    
+    # A flag which is set to 1 when a group needs to be removed from the op group table.
+    let f_dg = is_span * (group_count' - group_count)
+
+    # Enforce the constraint for updating op group table. The constraint specifies that when SPAN 
+    # or RESPAN operations are executed, we add between 1 and 7 groups to the op group table, and 
+    # when group count is decremented inside a span block, we remove a group from the op group 
+    # table.
+    enf p[3]' * (f_dg * u + 1 - f_dg) = 
+        p[3] * (get_f_g2(op_batch_flags) * v_1 + get_f_g4(op_batch_flags) * 
+        prod_v_3 + get_f_g8(op_batch_flags) * 
+        prod_v_7 - 1 + (f_span(op_bits) + get_f_respan(op_bits, extra)))
+    
+
+# Enforces proper decoding of span blocks.
+#
+# Constraint degree: 9
+ev span_block_constraints([addr, op_bits[7], hasher[8], is_span, group_count, op_index, op_batch_flags[3], s0, extra], [p[4]]):
+    enf inspan_column_constraints([op_bits, is_span, extra])
+    enf block_address_constraints([addr, is_span])
+    enf group_count_constraints([op_bits, hasher, is_span, group_count, extra])
+    enf op_group_decoding_constraints([op_bits, is_span, group_count, op_index, extra])
+    enf op_index_constraints([op_bits, is_span, group_count, op_index, extra])
+    enf op_batch_flag_constraints([op_bits, hasher, op_batch_flags, extra])
+    enf op_group_table_constraints([addr, op_bits, hasher, is_span, group_count, op_index, op_batch_flags, s0, extra], [p])
+
+
+### Decoder Air Constraints #######################################################################
+
+# Enforces the constraints on the decoder. The register `s0` denotes the value at the top of the 
+# stack. `extra` denotes the register for degree reduxtion during flags computation, and p[4] 
```

Comment:
nit: There are a few minor issues. It should be adjusted to: "... `extra` denotes the register for degree reduction during flag computations, ...